src/Compilers/CSharp/Portable/Lowering/NullCheckRewriter.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.CodeAnalysis.CodeGen;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.CSharp;
+
+namespace Microsoft.CodeAnalysis.CSharp.Lowering
+{
+    class NullCheckRewriter : BoundTreeRewriterWithStackGuardWithoutRecursionOnTheLeftOfBinaryOperator
+    {
+        readonly MethodSymbol _method;
+        SyntheticBoundNodeFactory _fact;
+        public NullCheckRewriter(
+            MethodSymbol method,
+            SyntaxNode syntax,
+            TypeCompilationState compilationState,
+            DiagnosticBag diagnostics)
+        {
+            _method = method;
+            _fact = new SyntheticBoundNodeFactory(method, syntax, compilationState, diagnostics);
+        }
+
+        internal static BoundNode Rewrite(
+            BoundStatement body,
+            MethodSymbol method,
+            int methodOrdinal,
+            VariableSlotAllocator slotAllocatorOpt,
+            TypeCompilationState compilationState,
+            DiagnosticBag diagnostics) //,
+                                       //out IteratorStateMachine stateMachineType)
```

Comment:
Was this comment left by accident?

Reply:
Removed but technically it was to remind me it would be needed for when I begin iterators.